sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -673,6 +677,244 @@ def test_check_estimator_get_tags_default_keys():
     )
 
 
+def test_check_classifiers_multilabel_output_format():
```

Comment:
Would it be a good idea for the individual estimator checks to also check the relevant estimator tags? For instance, the test shouldn't fail in this case if the tag is not set, and the output shape is wrong (or should it?)

Reply:
I am not totally sure. If the tag is not set then the check will not be run because we have a safety guard in the `yield_classifier_checks`.

However, I think that we are missing a test that check that the estimator are properly routed by `yield_***_checks` depending of their tags. I would probably do that in another PR (and we still need to refactor the tests).

Reply:
Naive comment about this test: as this test is quite long, would it be possible to break it in several ones?

Reply:
I would not break it now. I am going to refactor the API tests at some point. we could maybe we would like to have granular tests for `predict`/`predict_proba`/`decision_function` separately.